### PR TITLE
libvmi: init at 3e4114

### PR DESCRIPTION
**Pull request by proteansec**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libvmi/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  rev = "3e4114a64f012f1d3e2eb660bc65dcd130295d49";
+
+  src = fetchFromGitHub {
+    inherit rev;
+    owner = "libvmi";
+    repo = "libvmi";
+    sha256 = "0vbmrj0ij19i55afkqj64q7sgh0scpwk3c99qx6p6gn1qcy2wdss";
+  };
+
+  buildInputs = [ glib xen which yacc bison flex libtool autoconf automake pkgconfig libvirt ];
```
Comment (Mic92) [inline pkgs/development/libraries/libvmi/default.nix]:
```diff
- buildInputs = [ glib xen which yacc bison flex libtool autoconf automake pkgconfig libvirt ]; 
+ buildInputs = [ glib xen which libvirt ]; # probably which too.
+ nativeBuildInputs = [ yacc bison flex libtool autoconf automake pkgconfig ]; 
```

File: pkgs/development/libraries/libvmi/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, xen, which, fetchFromGitHub, yacc, bison, flex, glib, libtool, autoconf, automake, pkgconfig, libvirt }:
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  name = "libvirt-${version}";
+  version = "20170527-${stdenv.lib.strings.substring 0 7 rev}";
```
Comment (Mic92) [inline pkgs/development/libraries/libvmi/default.nix]: we prefer to have a numeric value at the end, to make it sortable for `nix-env -u`. See also https://nixos.org/nixpkgs/manual/#sec-package-naming

File: pkgs/development/libraries/libvmi/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    owner = "libvmi";
+    repo = "libvmi";
+    sha256 = "0vbmrj0ij19i55afkqj64q7sgh0scpwk3c99qx6p6gn1qcy2wdss";
+  };
+
+  buildInputs = [ glib xen which libvirt ];
+  nativeBuildInputs = [ yacc bison flex libtool autoconf automake pkgconfig ];
+
+  preConfigure = ''
+    ./autogen.sh
```
Comment (FRidh) [inline pkgs/development/libraries/libvmi/default.nix]: use `autoreconfHook` instead?

Comment (proteansec) [inline pkgs/development/libraries/libvmi/default.nix]: Thanks for the tip, fixed.

File: pkgs/development/libraries/libvmi/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, xen, which, autoreconfHook, fetchFromGitHub, yacc, bison, flex, glib, libtool, autoconf, automake, pkgconfig, libvirt }:
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  name = "libvirt-${version}";
```
Comment (Mic92) [inline pkgs/development/libraries/libvmi/default.nix]: Should be libvmi.

**Comment by Infinisil** [comment]
https://github.com/NixOS/nixpkgs/pull/42683

**Pull request merged**